src/compiler/checker.ts

```diff
@@ -7366,8 +7369,12 @@ namespace ts {
                 if (symbol && symbolIsValue(symbol)) {
                     return symbol;
                 }
-                if (resolved === anyFunctionType || resolved.callSignatures.length || resolved.constructSignatures.length) {
-                    const symbol = getPropertyOfObjectType(globalFunctionType, name);
+                const functionType = resolved === anyFunctionType ? globalFunctionType :
+                    resolved.callSignatures.length ? globalCallableFunctionType :

```

Comment:
Is there a reason we prefer callable to newable if a given type is both?

Reply:
No, not easily. We assume that all overloads are in a single list of signatures, so we'd somehow have to merge them or declare a _third_ type (`CallableNewableFunction`) in lib.d.ts that has the combined sets of overloads.